odc/stats/plugins/ls_ndvi_median.py

```diff
@@ -0,0 +1,104 @@
+"""
+Simple odc-stats plugin example, with excessive
+documentation. Creates temporal median NDVI
+from DEA Landsat Collection 3
+
+"""
+
+import xarray as xr
+from typing import Sequence, Tuple
+from datacube.utils.masking import mask_invalid_data
+from odc.stats.plugins._registry import register, StatsPluginInterface
+from odc.algo._masking import erase_bad, enum_to_bool
+
+
+class StatsNDVI(StatsPluginInterface):
+    """
+    Define a class for summarising time
+    series of NDVI using the median.
+    """
+
+    NAME = "ndvi_median"
+    SHORT_NAME = NAME
+    VERSION = "1.0"
+    PRODUCT_FAMILY = "ndvi"
+
+    def __init__(
+        self,
+        input_bands: Sequence[str] = None,
```

Comment:
I appreciate the type annotations in the example material.

The type annotations that assign None to non-optional types generate complaints by modern type checkers though, for pre-Python 3.10 the type should be `Optional[Sequence[str]]`, and for 3.10 and later one can use `Sequence[str] | None`.